samples/bluetooth/hci_rpmsg/CMakeLists.txt

```diff
@@ -2,6 +2,8 @@
 
 cmake_minimum_required(VERSION 3.13.1)
 
+list(APPEND OVERLAY_CONFIG ${CMAKE_CURRENT_SOURCE_DIR}/../hci_uart/hci_common.conf)
```

Comment:
I don't think this is very clean, if anyone has a better suggestion on where to place that file I'm all ears.

Reply:
Maybe drop `hci_uart` in the path.

@tejlmand Is there a better way to handle a common configuration overlay for multiple samples?

Reply:
No, not as of now.

And even this way is not acceptable.
When those settings are in `prj.conf`, then the user case decide to use a custom `prj.conf`, like `-DCONF_FILE=mysetting.conf`, in which case `prj.conf` will not be used.
Or user can provide custom Kconfig fragment with `-DOVERLAY_CONFIG=mysettings.conf`, but with the proposal in this PR, user cannot get rid of `hci_common.conf`.
In current form, it will always be appended.

I see the wish from maintenance view, that having a single file is nice, but for a user, it is actually nice to be able to simply look into `prj.conf` and see all the settings configured, and not having to figure out which additional files are used.

How the initial configuration is constructed is described here:
https://docs.zephyrproject.org/latest/guides/kconfig/setting.html#the-initial-configuration

This PR breaks what is described, so if something like this is needed, then a proper PR with such behavior should be opened, and docs updated.

The OVERLAY_CONFIG is described here:
https://docs.zephyrproject.org/latest/application/index.html#important-build-system-variables

Reply:
I agree, it's better for the user to not have to stitch together 20 different conf files :) So I will rework this PR to still get rid of all the duplicated conf items in the board-specific .conf , but will remove the common file and instead place the common items in their respective prj.conf (spi, uart etc) if you guys agree

Reply:
> but will remove the common file

so maybe update the PR description to reflect this :wink: